pkg/river/encoding/encoding.go

```diff
@@ -0,0 +1,148 @@
+package encoding
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	"github.com/grafana/agent/pkg/river/token/builder"
+
+	"github.com/grafana/agent/pkg/river/internal/rivertags"
+	"github.com/grafana/agent/pkg/river/internal/value"
+)
+
+const attr = "attr"
+const object = "object"
+
+// riverField is an interface that wraps the various concrete options for a river value.
+type riverField interface {
+	hasValue() bool
+}
+
+// ConvertRiverBodyToJSON is used to convertBase arguments, exports, health and debuginfo.
```

Comment:
Comment is stale and still mentions Flow-specific comments

```suggestion
// ConvertRiverBodyToJSON converts a River body value to a JSON representation.
```